templates/default/solr-healthcheck.sh.erb

```diff
@@ -1,6 +1,8 @@
 #!/bin/sh
 
 HEARTBEAT_FILE="/data/solr-healthcheck_<%= node['datashades']['hostname'] %>"
+# Only update heartbeat if it is present.
+# This allows us to manually drop a server from the pool
```

Comment:
can you add a script to add or remove a heart beat on the server we are on. need a bit more information on how this will pick a server and with the script running every minute could we have issues if we hit a slowdown etc.

Reply:
Manually removing a heartbeat shouldn't really need a script; just `rm` the file in `/data/`. I suppose it would be possible to write scripts that work for the server they're running on without needing the staff member to specify the hostname.

The current master is the alphabetically first server (considering only "solr1" through "solr9") that has a heartbeat no more than 2 minutes old.

Slowness shouldn't be a big problem here. The 'pick master' script just checks the contents of a small number of tiny files, without any external traffic, and the 'healthcheck' script just does a ping to localhost.

Reply:
if you know the system inside and out. yes, but for a new user having a script's like we do for or other systems is very helpful. Or a very beefy readme intro on this script with the scenarios in how it can be altered etc.

Reply:
I can add `heartbeat-on` and `heartbeat-off` scripts as the next step. There are several steps to go yet before this will enable proper failover.

Reply:
ok, then lets echo when its not found, that a touch to file (full path) is required for this to work again.

Reply:
This is meant to run via Cron. Any output would spam the root account with mail.